Sorting/HeapSort.cpp

```diff
@@ -0,0 +1,84 @@
+/*
+    Heap Sort :
+    Uses an array-implemented heap to sort a list of numbers.
+

```

Comment:
Change `numbers` to `integers` to be more specific.

Also add more information on what heapsort basically is/does (keep it short though). Something like:

> Heapsort is a comparison-based sorting algorithm that divides its input into a sorted and an unsorted region, and iteratively shrinks the unsorted region by extracting the largest element and moving that to the sorted region. It uses the heap data structure (implemented as a vector here) to find the maximum.